docs: add alternative ways to write a policy (#1513)
ash-project/ash: documentation/topics/security/policies.md

Each policy defined in a resource has two parts -

> - a condition or a list of conditions, such as `action_type(:read)` or `actor_attribute_equals(:admin, true)` or `always()`. If this condition is true for a given action request, then the policy will be applied to the request.

Reviewer: ```suggestion
- a condition or a list of conditions, such as `action_type(:read)`, `[action_type(:read), actor_attribute_equals(:admin, true)]` or `always()`. If the condition, or all conditions if given a list are true for a given action request, then the policy will be applied to the request.
```

If a single check does not explicitly authorize or forbid the whole policy, then the flow moves to the next check. For example, if an `authorize_if` check does NOT return true, this _does not mean the whole policy is forbidden_ - it means that further checking is required.

### Policy with `condition` inside `do` block

A condition or a list of conditions can also be moved inside the `policy` block.

> This way can make a really long list of conditions easier to read.

Reviewer: ```suggestion
This can make a really long list of conditions easier to read.
```

If a single check does not explicitly authorize or forbid the whole policy, then the flow moves to the next check. For example, if an `authorize_if` check does NOT return true, this _does not mean the whole policy is forbidden_ - it means that further checking is required.

### Alternative Ways To Write a Policy

#### Policy with keyword list

```elixir
policies do

>   policy always(), authorize_if: always()

Reviewer: I worry a bit about adding confusion for these various syntaxes. There isn't really any good reason for someone to have to know that you can use this keyword list syntax. The `do` block syntax is preferable. I think better to leave this out here, or at the very least put this at the bottom.

Author: I see. I did not know there was a preference toward  `do` block. I will remove it.

In my personal project, I was using mostly the keyword list variant, because I find it easier to read. Also it reduce some lines.

end
```

#### Policy with `condition` inside `do` block

> 

Reviewer: Let's add some explanation here. Like why they might want to specify the condition this way.

Putting the condition inside the policy block can make really long condition lists easier to read.

Author: I have added some explanation.

```elixir
policies do
  policy do

>     condition always()

Author: I was not aware of `condition`. I have discovered it in one of the [tests](https://github.com/ash-project/ash/blob/cef5900f40bed52b26a261ed30908faf3d016286/test/actions/aggregate_test.exs#L49).

>     condition(always())

Author: `mix spark.formatter` does not add `condition` to `.formatter.exs`.

Reviewer: I've fixed `spark.formatter` to include `condition`.

    authorize_if always()
  end
end
```

> #### Policy with multiple conditions

Author: I am not sure if it is the right place to mention multiple conditions for one policy.

```elixir
policies do

>   policy [condition1, condition2, condition3] do

Reviewer: I think it should be sufficient to change the original introduction of checks to contain the list example, which I mentioned above.

Author: I will remove it.
